crates/rpc/rpc/src/debug.rs

```diff
@@ -660,6 +660,15 @@ where
             .await
     }
 
+    pub async fn debug_code_by_hash(&self, hash: B256) -> RpcResult<Bytes> {
+        Ok(self
+            .provider()
+            .bytecode_by_hash(&hash)
```

Comment:
@mattsse this looks up bytecode via latest state provider, right? if so, we would want to do it using pending instead to capture newly created contracts

Reply:
usually this behaviour is derived from a Blockid parameter